lib/nimble_template/templates/template.ex

```diff
@@ -56,15 +58,28 @@ defmodule NimbleTemplate.Templates.Template do
     Mix.shell().cmd("npm install --prefix assets")
   end
 
-  defp suppress_necessary_credo_warnings(%Project{
-         base_module: base_module,
+  defp suppress_credo_warnings_for_base_project(%Project{base_module: base_module}) do
+    CredoHelper.disable_rule(
+      "lib/#{Macro.underscore(base_module)}.ex",
+      "CompassCredoPlugin.Check.DoSingleExpression"
+    )
+  end
+
+  defp suppress_credo_warnings_for_project_type(%Project{web_project?: true} = project) do
```

Comment:
What if this would be called with `mix_project?: true` or `live_project?: true`? Maybe we should add default matching function here

Reply:
If `live_project?: true` then `web_project` would also be true so it would apply the same for the web project so I didn't see any issue there, no difference of files etc.

But for `mix_project?: true` that's handled in `suppress_credo_warnings_for_base_project` which is also used by the others - it's common for all, just one file.

What about something like this what I had originally (more code though):

```elixir
defp post_apply!(%Project{mix_project?: true} = project) do
    order_dependencies!()
    fetch_and_install_elixir_dependencies()
    suppress_credo_warnings_for_project_type(project) # Re-use the below function now
    format_codebase()
  end

defp suppress_credo_warnings_for_project_type(%Project{mix_project?: true} = project) do
    suppress_credo_warnings_for_base_project(project)
end
```

Reply:
Just called each explicit removal directly with re-use happening inside each function instead of generic pattern matching function: 87d205bdc0d98879c094a7a7ca6b24c70509a5f3